bin/scanpy-find-variable-genes.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import logging
+import scanpy.api as sc
+from scanpy_wrapper_utils import ScanpyArgParser
+from scanpy_wrapper_utils import read_input_object
+from scanpy_wrapper_utils import write_output_object
+from scanpy_wrapper_utils import save_output_plot
+
+
+def main(args):
+    logging.debug(args)
+
+    adata = read_input_object(args.input_object_file, args.input_format)
+
+    min_mean, max_mean, min_disp, max_disp = None, None, None, None
+    for name, high, low in zip(args.parameter_names, args.high_thresholds, args.low_thresholds):
+        if name == 'mean':
+            min_mean = low
+            max_mean = high
+        elif name == 'disp':
+            min_disp = low
+            max_disp = high
+        else:
+            msg = 'Unsupported parameter name "{}", omitted'.format(name)
+            logging.warning(msg)
+
+    if args.output_plot:
+        sc.set_figure_params(format=args.output_plot_format)
+        filtered = sc.pp.filter_genes_dispersion(adata.X,
+                                                 flavor=args.flavor,
+                                                 min_mean=min_mean, max_mean=max_mean,
+                                                 min_disp=min_disp, max_disp=max_disp,
+                                                 n_bins=args.n_bins,
+                                                 n_top_genes=args.n_top_genes,
+                                                 log=(args.flavor == 'seurat'))
+        sc.settings.verbosity = 0
+        sc.pl.filter_genes_dispersion(filtered, show=False, save=True)
+        save_output_plot('filter_genes_dispersion', args.output_plot)
+
+    sc.pp.filter_genes_dispersion(adata,
+                                  flavor=args.flavor,
+                                  min_mean=min_mean, max_mean=max_mean,
+                                  min_disp=min_disp, max_disp=max_disp,
+                                  n_bins=args.n_bins,
+                                  n_top_genes=args.n_top_genes,
+                                  log=(args.flavor == 'seurat'))
+
+    write_output_object(adata, args.output_object_file, args.output_format)
+
+    logging.info('Done')
+    return 0
+
+
+if __name__ == '__main__':
+    argparser = ScanpyArgParser('Keep variable genes')
+    argparser.add_input_object()
+    argparser.add_output_object()
+    argparser.add_output_plot()
+    argparser.add_argument('--flavor',
```

Comment:
Shouldn't we use here `--dispersion-function` as we do in https://github.com/ebi-gene-expression-group/r-seurat-scripts/blob/856e595fd80ceb7c3f7d3e8f17dc25cf48c39fa8/seurat-find-variable-genes.R#L29 to try to keep interfaces similar?

Reply:
or is this not the same?

Reply:
They are related but not the same. Seurat allows different ways of calculating dispersion by `--dispersion-function`, whereas Scanpy doesn't. Instead, Scanpy gives two choices of normalising dispersion, either Seurat's method or Cell-Ranger's method, by `--flavor`.